src/common/com/intellij/plugins/haxe/ide/annotator/HaxeSemanticAnnotator.java

```diff
@@ -413,16 +442,83 @@ public void run() {
       holder.createErrorAnnotation(currentModifiers.getModifierPsi(HaxeModifierType.OVERRIDE), "Overriding nothing").registerFix(
         new RemoveModifierIntent("Remove override", currentModifiers, HaxeModifierType.OVERRIDE)
       );
-    } else if (!currentModifiers.hasModifier(HaxeModifierType.OVERRIDE) && requiredOverride) {
-      holder.createErrorAnnotation(currentMethod.getNameOrBasePsi(), "Must override").registerFix(
-        new HaxeFixer("Add override") {
-          @Override
-          public void run() {
-            currentModifiers.addModifier(HaxeModifierType.OVERRIDE);
+    }
+    else if (requiredOverride) {
+      if (!currentModifiers.hasModifier(HaxeModifierType.OVERRIDE)) {
+        holder.createErrorAnnotation(currentMethod.getNameOrBasePsi(), "Must override").registerFix(
+          new HaxeFixer("Add override") {
+            @Override
+            public void run() {
+              currentModifiers.addModifier(HaxeModifierType.OVERRIDE);
+            }
           }
-        }
+        );
+      }
+      else {
+        // It is rightly overriden. Now check the signature.
+        checkMethodsSignatureCompatibility(currentMethod, parentMethod, holder);
+      }
+    }
+  }
+
+  static public void checkMethodsSignatureCompatibility(
+    @NotNull final HaxeMethodModel currentMethod,
+    @NotNull final HaxeMethodModel parentMethod,
+    final AnnotationHolder holder
+  ) {
+    final HaxeDocumentModel document = currentMethod.getDocument();
```

Comment:
What I meant was that the check will not work correctly if the method being checked against (parent) contains optional arguments that are not included in the call being checked.  For example:

```haxe
  function doSomething(int:a, ?String:b, int:c) {}

  function doingSomethingelse() {
    doSomething(1,3);   //checkMethodSignatureCompatibilty will add "Non-matching arity expeced..." error.
  }
```
The error will be incorrect because the number of parameters check does not take into account that some of the parameters are optional.

Then the loop afterward, does not take into account that a type may match a later argument when the current argument is optional.

A solution may be to generate the list of correct signatures from the parent, and then see if the current signature matches any of them.  Then, if not find the closest matching one (the one that would mark the fewest errors?) and mark errors using that.